crates/circuit/src/dag_circuit.rs

```diff
@@ -6448,7 +6450,8 @@ impl DAGCircuit {
                 });
                 clbit_last_nodes
                     .entry(clbit)
-                    .and_modify(|val| *val = new_node);
+                    .and_modify(|val| *val = new_node)
+                    .or_insert(new_node);
```

Comment:
Was this a bug in `extend()` that you didn't catch until now?

Reply:
This one is not necessarily a bug since nodes should be there already. I believe the `.or_insert()` could be removed with no effect. I had it for debugging.